src/Symfony/Component/HtmlSanitizer/HtmlSanitizerConfig.php

```diff
@@ -0,0 +1,373 @@
+     *
+     * @var string[]|null
+     */
+    private ?array $allowedLinksHosts = null;
+
+    /**
+     * Should the sanitizer allow relative links (by default, they are dropped).
+     */
+    private bool $allowedLinksRelative = false;
+
+    /**
+     * Image/Audio/Video schemes that should be retained, other being dropped.
+     *
+     * @var string[]
+     */
+    private array $allowedMediaSchemes = ['http', 'https', 'data'];
+
+    /**
+     * Image/Audio/Video hosts that should be retained (by default, all hosts are allowed).
+     *
+     * @var string[]|null
+     */
+    private ?array $allowedMediaHosts = null;
+
+    /**
+     * Should the sanitizer allow relative media URL (by default, they are dropped).
+     */
+    private bool $allowedMediaRelative = false;
+
+    /**
+     * Should the URL in the sanitized document be transformed to HTTPS if they are using HTTP.
+     */
+    private bool $forceHttpsUrls = false;
+
+    /**
+     * Sanitizers that should be applied to specific attributes in addition to standard sanitization.
+     *
+     * @var Visitor\AttributeSanitizer\AttributeSanitizerInterface[]
+     */
+    private array $attributesSanitizers;
+
+    public function __construct()
+    {
+        $this->attributesSanitizers = [
+            new Visitor\AttributeSanitizer\AHrefSanitizer(),
+            new Visitor\AttributeSanitizer\MediaSrcSanitizer(),
+        ];
+    }
+
+    /**
+     * Create a configuration prefilled with allowed elements and attributes from the
+     * W3C Sanitizer API standard. {@see https://wicg.github.io/sanitizer-api/#default-configuration}.
+     */
+    public static function createStandardConfig(): self
+    {
+        $self = new self();
+
+        foreach (array_merge(W3CReference::BODY_ELEMENTS, W3CReference::HEAD_ELEMENTS) as $element) {
+            $self->allowedElements[$element] = [];
+
+            foreach (W3CReference::ATTRIBUTES as $attribute) {
+                $self->allowedElements[$element][$attribute] = true;
+            }
+        }
+
+        return $self;
+    }
+
+    /**
+     * Configure the URL schemes to be retained in href attributes of links. All other schemes
+     * will be dropped as they can lead to script execution.
+     *
+     * Calling this method will override all previous configuration for allowed links schemes.
+     */
+    public function withAllowedLinksSchemes(array $allowedLinksSchemes): self
+    {
+        $self = clone $this;
+        $self->allowedLinksSchemes = $allowedLinksSchemes;
+
+        return $self;
+    }
+
+    /**
+     * Configure the URL hosts to be retained in href attributes of links. All other hosts
+     * will be dropped. By default all hosts are allowed ($allowedLinksHosts = null).
+     *
+     * Calling this method will override all previous configuration for allowed links schemes.
+     */
+    public function withAllowedLinksHosts(?array $allowedLinksHosts): self
```

Comment:
let's try removing the with prefix as much as possible since not all methods use it already?
eg `allowLinksHosts()` here?
no `s` to `Link` btw? (same to other similar methods)